tests/functional/r/regression/regression_3507_typing_alias_isinstance.py

```diff
@@ -0,0 +1,24 @@
+"""
+https://github.com/PyCQA/pylint/issues/3507
+False-positive 'isinstance-second-argument-not-valid-type'
+for typing aliases in 'isinstance' calls.
+"""
+import collections
+import collections.abc
+import typing
+
+isinstance(42, typing.Dict)
+isinstance(42, typing.Counter)
+isinstance(42, typing.Collection)
+isinstance(42, typing.Iterator)
+isinstance(42, typing.Tuple)
+isinstance(42, typing.Callable)
```

Comment:
Maybe also typing.List and some other are missing ? Should this be exhaustive ?

Reply:
`typing.List` is similar to `typing.Dict` (which is tested). It wasn't my intention to add all possible aliases here. That would get out of hand quickly. Just a few selected ones.